src/parser/demonhunter/havoc/modules/talents/Demonic.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import TalentStatisticBox from 'interface/others/TalentStatisticBox';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import Events from 'parser/core/Events';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SpellLink from 'common/SpellLink';
+
+/**
+ * Example Report: https://www.warcraftlogs.com/reports/23dHWCrT18qhaJbz/#fight=1&source=16
+ */
+
+const META_BUFF_DURATION_EYEBEAM = 10000;
+
+class Demonic extends Analyzer{
+
+  eyeBeamCasts = 0;
+  goodDeathSweep = 0;
+  eyeBeamTimeStamp = undefined;
+  deathsweepsInMetaCounter = undefined;
+  badCasts = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.DEMONIC_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EYE_BEAM), this.onEyeBeamCast);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.DEATH_SWEEP), this.onDeathSweepCast);
+  }
+
+  onEyeBeamCast(event) {
+    const hasMetaBuff = this.selectedCombatant.hasBuff(SPELLS.METAMORPHOSIS_HAVOC_BUFF.id, event.timestamp - 1000);
+
+    if (hasMetaBuff) {
+      return;
+    }
+    this.eyeBeamCasts += 1;
+    this.eyeBeamTimeStamp = event.timestamp;
+
+    if (this.deathsweepsInMetaCounter === undefined) {
+      this.deathsweepsInMetaCounter = 0;
+      return;
+    }
+
+    if (this.deathsweepsInMetaCounter < 2) {
+      this.badCasts += 1;
+    }
+
+    this.deathsweepsInMetaCounter = 0;
+  }
+
+  onDeathSweepCast(event) {
+    if (this.eyeBeamTimeStamp !== undefined && (event.timestamp - this.eyeBeamTimeStamp) < META_BUFF_DURATION_EYEBEAM) {
+      this.goodDeathSweep += 1;
+      this.deathsweepsInMetaCounter += 1;
+    }
+  }
+
+  get avgDeathSweepPerEyeBeam(){
+    //Devide by 0 check incase they never cast Death Sweep
+    if(this.goodDeathSweep === 0){
+      return 0;
+    }
+    return this.goodDeathSweep / this.eyeBeamCasts;
+  }
+
+  get suggestionThresholds() {
+    return {
+      actual: this.badCasts,
+      isGreaterThan: {
+        minor: 0,
+        average: 0,
+        major: 1,
+      },
+      style: 'number',
+    };
+  }
+
+  suggestions(when) {
+    when(this.suggestionThresholds)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<>Try to have <SpellLink id={SPELLS.BLADE_DANCE.id} /> almost off cooldwon before casting <SpellLink id={SPELLS.EYE_BEAM.id} />. This will allow for two casts of <SpellLink id={SPELLS.DEATH_SWEEP.id} /> during the <SpellLink id={SPELLS.METAMORPHOSIS_HAVOC.id} />  buff you get from the <SpellLink id={SPELLS.DEMONIC_TALENT.id} /> talent.</>)
+          .icon(SPELLS.DEMONIC_TALENT.icon)
+          .actual(<>{actual} times during <SpellLink id={SPELLS.METAMORPHOSIS_HAVOC.id} /> <SpellLink id={SPELLS.DEATH_SWEEP.id} /> wasn't casted twice.</>)
```

Comment:
Again, never seen this here. Just make sure it works as intended, I've got no problem with it, if it works.

Reply:
Yup screenshot:
![image](https://user-images.githubusercontent.com/9600587/55111626-37604400-50b1-11e9-861d-d05894221cdf.png)